static/src/javascripts/projects/common/modules/atoms/audio-tracking.js

```diff
@@ -45,52 +44,47 @@ const record = (mediaId: string, eventType: string) => {
 const True = () => true;
 
 // Determines whether a reader has listened up to pct% of the audio track
-const percent = (pct: number) => (event: Event): boolean =>
-    event instanceof ProgressEvent &&
-    event.lengthComputable &&
-    Math.floor(event.loaded / event.total * 100) === pct;
+const percent = (pct: number) => (event: Event): boolean => {
+    const target: HTMLMediaElement = (event.target: any);
+    return Math.floor(target.currentTime / target.duration * 100) === pct;
```

Comment:
Can ` target.duration` ever be zero?

Reply:
hmm, I don't know to be honest, but since `x/0 = Infinity` in JS*, it won't be a problem.

* just writing it made me laugh 🤣